### PR TITLE
180 181 fix contact form

### LINKED ISSUE #180: add feedback when user submits contact form on /contact page
add success and error messages

### PR DESCRIPTION
**Pull request by sgnl**
resolves #180 & #181

### CODE REVIEW

**Comment by sgnl** [comment]
Sorry for the bad commit history, I was working and committing to `master` 😬

**Comment by sgnl** [comment]
error message:

image

**Comment by sgnl** [comment]
success message, form gets cleared via `.reset()`

image

File: compartments/assets/scripts/index.js [hunk 1]
```diff
@@ -5,22 +5,29 @@
...
-        return el.value === '' || !el.checkValidity() ? el.focus() : false;
+      var messageStatus = $('#display-message-status');
+
+      var formData = $(this).serializeArray().reduce(function (prev, curr) {
+        prev[curr.name] = curr.value;
+        return prev;
+      }, {});
+
+      // TODO replace with production url
+      $.post('//application.devleague/api/contact', formData).done(function () {
```
Comment (jaywon) [inline compartments/assets/scripts/index.js]: For the application server, now that SSL is setup we want to force `https`.

Server URL is: `https://application.devleague.com`

:bowing_man:

File: compartments/assets/scripts/index.js [hunk 2]
```diff
@@ -5,22 +5,29 @@
...
+        console.log('success!');
```
Comment (jaywon) [inline compartments/assets/scripts/index.js]: Please clean up `console.log`. :tada: for first emoji response message i've seen :1st_place_medal:

File: src/index.js [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
-    console.log('formData: ', formData);
-    alert('yay the form!')
-  };
+    const messageStatus = $('#display-message-status');
+
+    const formData = $(this).serializeArray().reduce((prev, curr) => {
+      prev[curr.name] = curr.value;
+      return prev;
+    }, {});
+
```
Comment (jaywon) [inline src/index.js]: Oops commented on `public/index.js`, messages apply here :man_facepalming:

**Pull request merged**